### PR TITLE
Fix va calculation

### PR DESCRIPTION
**Pull request by psolyca**
This PR has to be reviewed after #96.

This fix and enhanced the calculation of angles.

### CODE REVIEW

**Comment by steko** [comment]
I see a few commits that were already in #96 I think ?

**Comment by steko** [comment]
I hoped that these improvements would fix the long standing issue with Z coordinates in the RW5 format?

I'm referring to this test, marked as expected failure.

https://github.com/steko/totalopenstation/blob/6e68f0a798f2f3075888c93f176ea655c94a33cb/totalopenstation/tests/test_rw5.py#L15-L17

Perhaps we need to check with more data instead of just one point.

**Comment by psolyca** [comment]
> 
> 
> I see a few commits that were already in #96 I think ?

Yes, and I will rebase it on #96 after merge.
But it is for pre-review ^^

**Comment by psolyca** [comment]
Ok @steko , good for a review.

File: totalopenstation/tests/test_leica_gsi.py [hunk 1]
```diff
@@ -48,4 +48,4 @@ def test_linestring(self):
         self.ls = self.fp.build_linestring()
         self.assertAlmostEqual(self.ls.coords[0][0], 515.836)
         self.assertAlmostEqual(self.ls.coords[3][2], 2.553)
-        self.assertAlmostEqual(self.ls.bounds[0], -7.4662080)
+        self.assertAlmostEqual(self.ls.bounds[0], -7.4660913)
```
Comment (steko) [inline totalopenstation/tests/test_leica_gsi.py:51]: This is a slight change, but do we have a way to check what it the correct value without adjusting tests? E.g. with known sample data that is processed by another "trusted" software.

File: totalopenstation/tests/test_nikon.py [hunk 2]
```diff
@@ -17,8 +17,8 @@ def test_points(self):
     def test_basepoint(self):
         self.assertAlmostEqual(self.fp.points[0].geometry.x, 0.0)
         self.assertAlmostEqual(self.fp.points[0].geometry.y, 0.0)
-        self.assertAlmostEqual(self.fp2.points[7].geometry.x, 9936.99390187)
-        self.assertAlmostEqual(self.fp2.points[7].geometry.y, 10036.06473563)
+        self.assertAlmostEqual(self.fp2.points[7].geometry.x, 10000.08905808)
```
Comment (steko) [inline totalopenstation/tests/test_nikon.py:20]: This is a slight change, but do we have a way to check what it the correct value without adjusting tests? E.g. with known sample data that is processed by another "trusted" software.
No need to change here however!

**Review by steko — APPROVED** [review]
Ok, it seems the previous version of polar point conversion (mostly my work) was rather bad, excellent to see all these improvements! 👍

File: totalopenstation/tests/test_leica_gsi.py (hunk 1 above)
Comment (psolyca) [reply totalopenstation/tests/test_leica_gsi.py:51]: I do not know what is your expectation on precision.
I think mm for metric is the minimum and it is really difficult to obtain a precision less than that during a survey or stakeout.

**Pull request merged**